src/logic/safe/store/actions/createTransaction.ts

```diff
@@ -220,19 +221,26 @@ export class TxSender {
     confirmCallback?: ConfirmEventHandler,
     errorCallback?: ErrorEventHandler,
   ): Promise<string | undefined> {
-    const isOffchain = await this.canSignOffchain(state)
+    // Confirmation (on-/off-chain signing)
+    if (!this.isFinalization) {
+      let signature: undefined | string = undefined
+      let isValidSignature = false
+
+      const canSignOffChain = this.canSignOffchain(state)
 
-    // Off-chain signature
-    if (!this.isFinalization && isOffchain) {
       try {
-        const { hardwareWallet } = providerSelector(state)
-        const signature = await this.onlyConfirm(hardwareWallet)
+        if (canSignOffChain) {
+          const { hardwareWallet } = providerSelector(state)
+          signature = await this.getOffChainSignature(hardwareWallet)
+          isValidSignature = !!signature
+        } else {
+          isValidSignature = await isValid1271Signature(this.txArgs.sender, this.safeTxHash)
```

Comment:
It won't work for Argent because they return an ECDSA signature that needs to be validated via the  EIP1271 method

Reply:
Just ping me any time :)

Another question popped up in my mind: why do we want to verify the EIP-1271 signature instead of simply proceeding with them and letting the backend do the validation? The same way we do with regular ECDSA signatures, we don't validate them; the interface forwards them to the backend

Reply:
This was just an investigative Friday project on my behalf. I'll close this now.